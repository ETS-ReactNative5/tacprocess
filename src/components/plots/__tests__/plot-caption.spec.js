import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount, render } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import PlotCaption from './PlotCaption'

const caption ='The Plot Caption'

// Checking if the PlotCaption Component renders correctly for different inputs
describe('PlotCaption Component', () => {

  // Enzyme method testing if it renders correctly with empty value of the target
  it('Render the PlotCaption Component having unpopulated props with no errors', () => {
    const rendered = mount(<PlotCaption caption='' />)
    expect(shallowToJson(rendered)).toMatchSnapshot()
  })

  it('Render the PlotCaption Component having populated props with no errors', () => {
    const rendered = mount(<PlotCaption caption={ caption } />)
    expect(shallowToJson(rendered)).toMatchSnapshot()
  })

})
