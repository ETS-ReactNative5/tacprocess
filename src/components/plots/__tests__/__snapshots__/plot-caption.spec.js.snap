// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlotCaption Component Render the PlotCaption Component having populated props with no errors 1`] = `
<PlotCaption
  caption="The Plot Caption"
>
  <div
    className="plot-caption"
    dangerouslySetInnerHTML={
      Object {
        "__html": "The Plot Caption",
      }
    }
  />
</PlotCaption>
`;

exports[`PlotCaption Component Render the PlotCaption Component having unpopulated props with no errors 1`] = `
<PlotCaption
  caption=""
>
  <div
    className="plot-caption"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</PlotCaption>
`;
